source/wp-content/themes/wporg-developer-2023/src/cli-command-table/block.php

```diff
@@ -0,0 +1,75 @@
+<?php
+namespace WordPressdotorg\Theme\Developer_2023\Dynamic_CLI_Command_Table;
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function init() {
+	register_block_type(
+		dirname( dirname( __DIR__ ) ) . '/build/cli-command-table',
+		array(
+			'render_callback' => __NAMESPACE__ . '\render',
+		)
+	);
+}
+
+/**
+ * Render the block content.
+ *
+ * @return string Returns the block markup.
+ */
+function render() {
+	$posts = get_posts(
+		array(
+			'post_type'      => 'command',
+			'order'          => 'ASC',
+			'post_parent'    => 0,
+			'nopaging'       => true,
+		)
+	);
+
+	if ( is_wp_error( $posts ) || empty( $posts ) ) {
```

Comment:
I think this is right but I can't find any documentation as to whether `is_wp_error( $posts )` should be checked. I've just seen it around the code base in similar scenarios.